cmd/collector/app/span_processor.go

```diff
@@ -178,10 +178,21 @@ func (sp *spanProcessor) processItemFromQueue(item *queueItem) {
 }
 
 func (sp *spanProcessor) addCollectorTags(span *model.Span) {
-	// TODO add support for deduping tags, https://github.com/jaegertracing/jaeger/issues/1778
+	dedupKey := make(map[string]struct{})
```

Comment:
please add an early exit so that people who do not enrich tags in the collector won't pay the perf penalty
```go
if len(sp.collectorTags) == 0 {
    return
}
```

Reply:
@yurishkuro Done